src/pudl/extract/eiaaeo.py

```diff
@@ -438,21 +444,28 @@ def raw_eiaaeo(context: AssetExecutionContext):
     }
     ds = context.resources.datastore
 
-    # TODO (daz 2024-04-15): one day, we might want the AEO for more than one
-    # year. But for now we only take the first year from the settings.
-    year = context.resources.dataset_settings.eia.eiaaeo.years[0]
-    filename = f"AEO{year}.txt"
-
-    with ds.get_zipfile_resource("eiaaeo", year=year).open(
-        filename, mode="r"
-    ) as aeo_raw:
-        taxonomy = AEOTaxonomy(io.TextIOWrapper(aeo_raw))
+    # Extract for all years specified in the settings
+    years = context.resources.dataset_settings.eia.eiaaeo.years
 
     selected = context.op_execution_context.selected_output_names
     for asset_name in selected:
-        yield Output(
-            value=taxonomy.get_table(name_to_number[asset_name]), output_name=asset_name
-        )
+        all_years_data = []
+        for year in years:
+            filename = f"AEO{year}.txt"
+            with ds.get_zipfile_resource("eiaaeo", year=year).open(
+                filename, mode="r"
+            ) as aeo_raw:
+                taxonomy = AEOTaxonomy(io.TextIOWrapper(aeo_raw))
```

Comment:
tricky -- it looks like this will read each AEO file multiple times, once for each selected asset. We have more assets than years right now, so that's not great.

But if we flip the order of the loops, so that we go through years in the outer loop and assets in the inner loop, we'll need to hold a bunch of stuff in memory until all the years have been processed before we can yield the outputs. Also not great.

What's your sense of how resource-intensive this is? if it's slow, or a memory hog, we can look at that tradeoff more closely, but if it's speedy and tiny then maybe it doesn't matter and we can stick with what's easiest to read